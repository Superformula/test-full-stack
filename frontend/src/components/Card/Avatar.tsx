import styles from "./Card.module.css";
import { useEffect, useState } from "react";
import { createApi } from "unsplash-js";

const api = createApi({
  // Don't forget to set your access token here!
  // See https://unsplash.com/developers
  accessKey: "M1dsgksncu-N8pjPvgshjGXU6tRjhWjpqIfD28rtyg0",
});

const Avatar = () => {
  const [data, setPhotosResponse] = useState(null);

  useEffect(() => {
    api.photos
      .getRandom({ query: "portrait", orientation: "portrait" })
      .then((result) => {
        setPhotosResponse(result);
      })
      .catch(() => {
        console.log("something went wrong!");
      });
  }, []);

  console.log(data);

  if (!data) {
    return null;
  }

  return <img className={styles.avatar} src={data.response.urls.regular} />;
};

export default Avatar;
