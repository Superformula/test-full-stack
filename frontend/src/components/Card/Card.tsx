import React from 'react';
import moment from 'moment';

import styles from './Card.module.scss';
import { User } from '../../types';

type Props = {
  user: User;
  onClick: () => void;
};

const parseDate = (date?: string) => moment(date).format('DD MMM YYYY');

const Card: React.FC<Props> = ({ user, onClick }) => (
  <div className={styles.wrapper} onClick={onClick}>
    <img
      className={styles.avatar}
      src={`https://source.unsplash.com/random/168x168?sig=${user.id}`}
    ></img>
    <div className={styles.middleInfo}>
      <div className={styles.name}>{user.name}</div>
      <div>
        {user.updatedAt ? (
          <>
            <span className={styles.dateText}>updated</span>
            <span className={styles.dateValue}>
              {parseDate(user.updatedAt)}
            </span>
          </>
        ) : (
          <>
            <span className={styles.dateText}>created</span>
            <span className={styles.dateValue}>
              {parseDate(user.createdAt)}
            </span>
          </>
        )}
      </div>
    </div>
    <div className={styles.description}>{user.description}</div>
  </div>
);

export default Card;
