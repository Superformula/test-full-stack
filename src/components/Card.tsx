import React, { Component } from "react";
import "../assets/App.scss";

import editIcon from "../assets/edit-icon.svg";
import EditModal from "./EditModal";
import { User } from "../models";

type CardProps = {
  key: Number;
  user: User;
};

class Card extends Component<CardProps, any> {
  constructor(props: CardProps) {
    super(props);
    this.state = {
      showModal: false,
      createdAt: new Date(props.user.createdAt).toLocaleDateString("en-US", {
        day: "2-digit",
        month: "short",
        year: "numeric",
      }),
    };
  }

  closeModal = () => this.setState({ showModal: false });
  openModal = () => this.setState({ showModal: true });

  render() {
    return (
      <div className="Card" onClick={this.openModal}>
        <img className="Card-edit-button" src={editIcon} alt="Edit User Icon" />
        <div className="Card-image">
          <img src={this.props.user.avatar.toString()} alt="User Name Avatar" />
        </div>
        <div className="Card-title">
          <h3>{this.props.user.name}</h3>
          <div className="Card-date">
            created <span>{this.state.createdAt}</span>
          </div>
        </div>
        <div className="Card-description">
          <p>{this.props.user.description}</p>
        </div>
        <EditModal
          showModal={this.state.showModal}
          handleClose={this.closeModal}
          data={{
            id: this.props.user.id,
            name: this.props.user.name,
            address: this.props.user.address,
            description: this.props.user.description,
          }}
        />
      </div>
    );
  }
}

export default Card;
