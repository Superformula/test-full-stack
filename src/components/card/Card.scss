@import "src/styles/mixins";

.card {
  width: 100%;

  @include desktop {
    width: 400px;
  }

  &:hover {
    box-shadow: 0 4px 10px rgba(0,0,0,.1);
  }
}
