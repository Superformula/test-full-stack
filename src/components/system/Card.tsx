import styled from "@emotion/styled";

export const Card = styled.button`
  align-items: center;
  background-color: ${({ theme }) => theme.color.bgPrimary};
  border: none;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  height: 21rem;
  justify-content: space-between;
  padding: 2.5rem 2rem;
  text-align: left;
  text-decoration: none;
  transition: box-shadow 0.3s ease-in-out;
  width: 25rem;

  :hover {
    box-shadow: 0px 4px 10px ${(props) => props.theme.color.divider};
  }
  :focus {
    outline: none;
    box-shadow: 0 0 0 4px ${({ theme }) => theme.color.focus};
  }
`;

export const CardTitle = styled.h2`
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
`;

export const CardSubtitle = styled.p`
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const CardContent = styled.div`
  align-self: stretch;
`;
