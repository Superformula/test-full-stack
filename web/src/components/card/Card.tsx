import React, { ReactElement, ReactNode } from 'react'
import style from './Card.module.scss'

interface CardProps {
  children: ReactNode
}

export const Card = ({ children }: CardProps): ReactElement => <div className={style.card}>{children}</div>
