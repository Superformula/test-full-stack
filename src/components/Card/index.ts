import Card from "./Card"
export type { User } from "./Card"

export default Card;
