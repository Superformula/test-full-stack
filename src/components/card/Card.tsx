import React from 'react';
import classnames from 'classnames';
import './Card.scss';

interface CardProps {
  className?: string
}

export const Card: React.FC<CardProps> = ({ children, className }) => (
  <div className={classnames('card', className)}>{children}</div>
);
