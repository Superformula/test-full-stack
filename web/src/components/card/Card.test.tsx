import { render } from '@testing-library/react'
import React from 'react'
import { Card } from './Card'

test('renders card and matches snapshot', () => {
  const { container } = render(<Card>Hello World</Card>)
  expect(container).toMatchSnapshot()
})
