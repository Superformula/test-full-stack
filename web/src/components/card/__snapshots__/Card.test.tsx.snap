// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card and matches snapshot 1`] = `
<div>
  <div
    class="card"
  >
    Hello World
  </div>
</div>
`;
